Library/Homebrew/test/test_cmd_testbot.rb

```diff
@@ -0,0 +1,44 @@
+require "testing_env"
+require "dev-cmd/test-bot"
+
+class TestbotCommandTests < Homebrew::TestCase
+  def with_environment(h)
+    old = Hash[h.keys.map { |k| [k, ENV[k]] }]
+    ENV.update h
+    begin
+      yield
+    ensure
+      ENV.update old
+    end
```

Comment:
You might want to use `ENV.to_hash` and `ENV.replace` for backing up and restoring the environment. This would also improve consistency with other tests, e.g. [these lines from `test_utils.rb`](https://github.com/Homebrew/brew/blob/cab97cf4d782ce36762f0cd615ca59718f55b0d5/Library/Homebrew/test/test_utils.rb#L36-L44), and I believe also other environment handling code we have in Homebrew.